FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Models/FhirSearch/TypeFilter.cs

```diff
@@ -0,0 +1,39 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Synapse.Common.Models.FhirSearch
+{
+    public class TypeFilter
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TypeFilter"/> class.
+        /// We will trigger a http request for each typeFilter,
+        /// </summary>
+        /// <param name="resourceType">resource type</param>
+        /// <param name="parameters">query parameters</param>
+        public TypeFilter(string resourceType, IList<Tuple<string, string>> parameters)
+        {
+            ResourceType = resourceType;
+            Parameters = parameters ?? new List<Tuple<string, string>>();
+        }
+
+        /// <summary>
+        /// Resource type, if the resourceType is "*" means want to get all the compartment resources.
+        /// </summary>
+        [JsonProperty("resourceType")]
+        public string ResourceType { get; set; }
```

Comment:
Will it make more sense to use enum type instead of string for `ResourceType`? If the `ResourceType` has fixed range of values which can be fetched from some FHIR libraries.

Reply:
The `ResourceType` is different between `R4` and `STU3`, it is not immutable. 
And we use [HL7.Fhir.Model](https://docs.fire.ly/projects/Firely-NET-SDK/index.html) to parse and validate resource types, it supports returning a string list of the supported resource types and validating if a string is a known resource.

So, I think the string type is more suitable here.

Additionally, FHIR server also uses string type for resource type in [ExportJobFilter](https://github.com/microsoft/fhir-server/blob/c7ff42bf498e4c38ab18d8b17fc0a0131a366444/src/Microsoft.Health.Fhir.Core/Features/Operations/Export/Models/ExportJobFilter.cs#L30)

Reply:
Got it, thanks.